src/main/java/hudson/plugins/promoted_builds/Status.java

```diff
@@ -106,19 +106,24 @@ public PromotionProcess getProcess() {
     public String getIcon(String size) {
         String baseName;
 
-        PromotionProcess p = getProcess();
-        if (p == null) {
-            // promotion process undefined (perhaps deleted?). fallback to the default icon
-            baseName = "star-gold";
-        } else {
-            Promotion l = getLast();
-            if (l!=null && l.getResult()!= Result.SUCCESS)
-                return Jenkins.RESOURCE_PATH+"/images/"+size+"/error.png";
-
-            baseName = p.getIcon();
-        }
-        return Jenkins.RESOURCE_PATH+"/plugin/promoted-builds/icons/"+size+"/"+ baseName +".png";
-    }
+		PromotionProcess p = getProcess();
+		if(p == null) {
+			// promotion process undefined (perhaps deleted?). fallback to the default icon
+			baseName = "star-gold";
+		}
+		else {
+			Promotion l = getLast();
+			if(l != null && l.getResult() != Result.SUCCESS)
+				return Jenkins.RESOURCE_PATH + "/images/" + size + "/error.png";
+
+			baseName = l.getIcon();
+			if(baseName == null) {
+				baseName = p.getIcon();
+
+			}
+		}
+		return baseName != null && !"none".equals(baseName) ? Jenkins.RESOURCE_PATH + "/plugin/promoted-builds/icons/" + size + "/" + baseName + ".png" : null;
```

Comment:
Since the original method never returns null, currently there is no way to remove a badge that was already added by the promotion process. 

We use the promoted builds plugin to both promote as well as demote a build. If the user wants to demote a build, there is currently no way to remove the promotion badge. 

with the above enhancement, we can add a groovy build step to set the build's icon to an empty string to remove the badge if the user decides to demote an already promoted build.

The only time this method returns null is when someone explicitly sets the build's icon to an empty string. So this change will always be backwards compatible.